tests/js/server/aql/aql-optimizer-keep.js

```diff
@@ -80,6 +80,16 @@ function optimizerKeepTestSuite () {
       assertEqual([ "test0", "test1", "test2", "test3", "test4", "test5", "test6", "test7", "test8", "test9" ], results.json, query);
     },
 
+////////////////////////////////////////////////////////////////////////////////
+/// @brief test refer to variable introduced by COLLECT from KEEP
+////////////////////////////////////////////////////////////////////////////////
+    
+    testReferToCollectVariablesFromKeep : function () {
+      assertQueryError(errors.ERROR_QUERY_PARSE.code, "FOR i IN " + c.name() + " COLLECT a = i KEEP a RETURN 1");
+      assertQueryError(errors.ERROR_QUERY_PARSE.code, "FOR i IN " + c.name() + " COLLECT a = i KEEP i RETURN 1");
```

Comment:
I'm confused by the second line. Shouldn't it be OK to `KEEP i`, as it was introduced previously by `FOR`? Even though it doesn't make much sense to both `COLLECT` and `KEEP` it. What am I overlooking?